sda-commons-server-auth-testing/src/test/java/org/sdase/commons/server/auth/testing/KeyLoaderProxyIT.java

```diff
@@ -38,35 +39,50 @@
 class KeyLoaderProxyIT {
   @RegisterExtension
   @Order(0)
-  static final WireMockClassExtension PROXY_WIRE =
-      new WireMockClassExtension(wireMockConfig().dynamicPort());
+  static final WireMockExtension PROXY_WIRE =
+      new WireMockExtension.Builder().options(wireMockConfig().dynamicPort()).build();
 
   @RegisterExtension
   @Order(1)
   static final SystemPropertyClassExtension PROP =
       new SystemPropertyClassExtension()
-          .setProperty("http.proxyPort", () -> "" + PROXY_WIRE.port());
+          .setProperty("http.proxyPort", () -> "" + PROXY_WIRE.getPort());
 
-  @RegisterExtension
-  @Order(2)
-  static final DropwizardAppExtension<AuthTestConfig> DW =
-      new DropwizardAppExtension<>(
-          AuthTestApp.class,
-          ResourceHelpers.resourceFilePath("test-config.yaml"),
-          config("auth.keys[0].type", "JWKS"),
-          config("auth.keys[0].location", "http://sda.se/jwks"));
+  static DropwizardAppExtension<AuthTestConfig> DW;
 
   @BeforeAll
-  static void beforeAll() {
+  static void beforeAll() throws Exception {
     // expect that the proxy receives the request
     PROXY_WIRE.stubFor(
         get("/jwks")
             .withHeader(HttpHeaders.HOST, equalTo("sda.se"))
             .willReturn(okJson("{\"keys\": []}")));
   }
 
+  @AfterAll
+  static void afterAll() {
+    DW.after();
+  }
+
   @Test
-  void shouldUseProxy() {
+  void shouldUseProxy() throws Exception {
+
+    //    instantiate DropwizardAppExtension here to avoid a race condition with the
+    // WireMockExtension
+
+    //    WireMockExtension#beforeEach(ExtensionContext context) resets the logged calls.
+    //    during initialization of DropwizardAppExtension the JwksKeySource#loadKeysFromSource()
+    // method calls wiremock stub
+
+    DW =
```

Comment:
Could it be initialized in `@BeforeAll`?

If it's initialized here in the test, I would also expect to shut it it down in a finally block and not in `@AfterAll` for consistency.

Reply:
No, local tests show that the race condition still exists.

Moved shut down to finally block
https://github.com/SDA-SE/sda-dropwizard-commons/pull/2973